lib/berkshelf/source.rb

```diff
@@ -9,8 +9,18 @@ class Source
 
     # @param [String, Berkshelf::SourceURI] uri
     def initialize(uri)
-      @uri        = SourceURI.parse(uri)
-      @api_client = APIClient.new(uri, ssl: {verify: Berkshelf::Config.instance.ssl.verify})
+      if uri == :chef_server
+        @uri = SourceURI.parse(Berkshelf::Config.instance.chef.chef_server_url)
+        @api_client = APIClient.chef_server(
+          ssl: {verify: Berkshelf::Config.instance.ssl.verify},
+          client_name: Berkshelf::Config.instance.chef.node_name,
+          server_url: Berkshelf::Config.instance.chef.chef_server_url,
+          client_key: Berkshelf::Config.instance.chef.client_key,
+        )
+      else
+        @uri        = SourceURI.parse(uri)
+        @api_client = APIClient.new(uri, ssl: {verify: Berkshelf::Config.instance.ssl.verify})
```

Comment:
maybe you could move these into lazy accessors?

```
def api_client
  @api_client ||= begin
    if uri == :chef_server
      APIClient.chef_server(
           ssl: {verify: Berkshelf::Config.instance.ssl.verify},
           client_name: Berkshelf::Config.instance.chef.node_name,
           server_url: Berkshelf::Config.instance.chef.chef_server_url,
           client_key: Berkshelf::Config.instance.chef.client_key,
         )
    else
      APIClient.new(uri, ssl: {verify: Berkshelf::Config.instance.ssl.verify})
    end
  end
end
```

and similar for `def uri`

ah, yeah, and the parameter to initialize should also probably be called "source" and you should stuff that into an `attr_accessor :source` and then use that to lazy build the uri from it...